docs/adrs/007-minifront-v2-architecture.md

```diff
@@ -0,0 +1,261 @@
+- Improved testability through dependency injection
+- Reusable business logic across stores
+- Easier mocking for testing
+
+#### 4. Dependency Injection Pattern
+
+**New addition**: Root store pattern with dependency injection
+
+```typescript
+export class RootStore {
+  readonly balancesStore: BalancesStore;
+  readonly transactionsStore: TransactionsStore;
+
+  constructor(private penumbraService: PenumbraService) {
+    this.balancesStore = new BalancesStore(this, penumbraService);
+    this.transactionsStore = new TransactionsStore(this, penumbraService);
+  }
+}
+```
+
+**Benefits:**
+
+- Centralized configuration and initialization
+- Cross-store communication through shared root
+- Easier testing with mock services
+- Clear application lifecycle management
+
+### Technical Stack Evolution
+
+| Aspect               | Original Minifront           | Minifront v2            | Reasoning                                            |
+| -------------------- | ---------------------------- | ----------------------- | ---------------------------------------------------- |
+| **Routing**          | React Router (hash)          | React Router (hash)     | ✅ **Parity**: Maintains deployment flexibility      |
+| **Build Tool**       | Vite                         | Vite                    | ✅ **Parity**: Continues fast development experience |
+| **State Management** | Zustand + TanStack Query     | MobX                    | 🔄 **Evolution**: Better performance & DX            |
+| **Architecture**     | Component-based              | Feature-Sliced Design   | 🔄 **Evolution**: Improved scalability               |
+| **Services**         | Direct in components         | Dedicated service layer | ➕ **Addition**: Better separation of concerns       |
+| **UI Library**       | @penumbra-zone/ui-deprecated | @penumbra-zone/ui       | 🔄 **Migration**: Complete UI library change         |
+| **CSS Framework**    | Tailwind                     | Tailwind                | ✅ **Parity**: Consistent styling approach           |
+| **TypeScript**       | TypeScript                   | TypeScript              | ✅ **Parity**: Type safety maintained                |
+
+### UI Component Strategy
+
+**Major divergence**: Complete migration from `@penumbra-zone/ui-deprecated` to `@penumbra-zone/ui`
+
+```typescript
+// Original minifront (ui-deprecated)
+import { Button } from '@penumbra-zone/ui-deprecated/components/ui/button';
+import { Card } from '@penumbra-zone/ui-deprecated/components/ui/card';
+import '@penumbra-zone/ui-deprecated/styles/globals.css';
+
+// Minifront-v2 (new ui library)
+import { Button } from '@penumbra-zone/ui/Button';
+import { Card } from '@penumbra-zone/ui/Card';
+import '@penumbra-zone/ui/style.css';
+
+// Plus v2-specific UI components
+import { TransactionCard } from '@/shared/ui/transaction-card';
+```
+
+This represents a **complete UI library migration**, not preservation of existing patterns.
+
+## Rationale
```

Comment:
nitpick: too much AI 😅

I mean, if the text is generated fast, why should we spend time reading it? Let's keep it short and concise please